rust/tw_hash/src/blake2.rs

```diff
@@ -4,26 +4,65 @@
 // terms governing use, modification, and redistribution, is contained in the
 // file LICENSE at the root of the source code distribution tree.
 
+use crate::Error;
 use blake2::{
     digest::{Update, VariableOutput},
     Blake2bVar,
 };
 use blake2b_ref::Blake2bBuilder;
```

Comment:
Why pull `blake2b_ref` when `blake2` itself supports `no_std`? Also, it's not quite clear to me what `Blake2bBuilder::personal(...)` does, is it context data (and is there an equivalent to it in `blake2`)?

Reply:
Not sure if I understood the question correctly. We import `Blake2Builder` in order to specify the `personal`. But I think we could replace `blake2::Blake2bVar` with `blake2b_ref::Blake2Builder`.
> Why pull blake2b_ref when blake2 itself supports no_std? 

As I can see, the personal is used to personalize some blockchain hashes (avoid collision? 🤔 ). For example:
https://github.com/trustwallet/wallet-core/blob/d67078daa580b37063c97be66a625aaee9664882/src/Nervos/Address.cpp#L122

Reply:
Done
Replaced `blake2` dependency with `blake2b_ref`